zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -0,0 +1,727 @@
+use crate::{
+    evm_circuit::{
+        param::{STEP_HEIGHT, STEP_WIDTH},
+        step::{ExecutionState, Preset, Step},
+        table::{FixedTableTag, Lookup, LookupTable, Table},
+        util::constraint_builder::ConstraintBuilder,
+    },
+    util::Expr,
+};
+use halo2::{
+    arithmetic::FieldExt,
+    circuit::{Layouter, Region},
+    plonk::{
+        Column, ConstraintSystem, Error, Expression, Fixed, Instance, Selector,
+    },
+    poly::Rotation,
+};
+use std::collections::HashMap;
+
+mod add;
+mod bitwise;
+mod byte;
+mod comparator;
+mod dup;
+mod error_oog_pure_memory;
+mod jump;
+mod jumpdest;
+mod jumpi;
+mod memory;
+mod pc;
+mod pop;
+mod push;
+mod signextend;
+mod stop;
+mod swap;
+use add::AddGadget;
+use bitwise::BitwiseGadget;
+use byte::ByteGadget;
+use comparator::ComparatorGadget;
+use dup::DupGadget;
+use error_oog_pure_memory::ErrorOOGPureMemoryGadget;
+use jump::JumpGadget;
+use jumpdest::JumpdestGadget;
+use jumpi::JumpiGadget;
+use memory::MemoryGadget;
+use pc::PcGadget;
+use pop::PopGadget;
+use push::PushGadget;
+use signextend::SignextendGadget;
+use stop::StopGadget;
+use swap::SwapGadget;
+
+#[allow(missing_docs)]
+pub mod bus_mapping_tmp {
+    use crate::evm_circuit::{
+        step::ExecutionState,
+        table::{RwTableTag, TxContextFieldTag},
+        util::RandomLinearCombination,
+    };
+    use bus_mapping::{
+        eth_types::{Address, ToLittleEndian, ToScalar, Word},
+        evm::OpcodeId,
+    };
+    use halo2::arithmetic::FieldExt;
+    use sha3::{Digest, Keccak256};
+
+    #[derive(Debug, Default)]
+    pub struct Block<F> {
+        // randomness for random linear combination
+        pub randomness: F,
+        pub txs: Vec<Transaction<F>>,
+        pub rws: Vec<Rw>,
+        pub bytecodes: Vec<Bytecode>,
```

Comment:
do you plan to add block related properties (block number,  difficulty ,  coinbase, timestamp .etc.)  to support block relevant op codes?

Reply:
yeah, but I don't think I should keep adding things in this PR, I can add this right after this PR and the changes should be easier to review. Do you think it's okay to add in next PR? Or is this blocking any current progress?

Reply:
I am reviewing those op codes about block and planning to implement ,  I can add this part later.

Reply:
I am reviewing those op codes about block and planning to implement ,  I can add this part later.

Reply:
Thanks, that will be great :)